### ISSUE TITLE
Parallel PettingZoo Env & Multi-agent Policies

### ISSUE THREAD

**Issue by JenniferHahn** [LucasAlegre/sumo-rl#151]
I wanted to open this thread again since I am facing similar issues - I am trying to run multi-agent RL with a parallel petting zoo env which I am wrapping using superset (which creates the largest obs space for all) but I would actually like to have different policies for every agent since the action and observation space is initially different:

register_env(env_name, lambda config: ParallelPettingZooEnv(env_creator(config)))
env = PettingZooEnv(env_creator({}))
...
within the config I would like to set
        .multiagent(
            policies={id: (PPOTF1Policy, env.observation_spaces[id], env.action_spaces[id], {}) for id in env.agents},
            policy_mapping_fn= (lambda id: id) 
        )

Is there any way to achieve both - use the (wrapped) environment but implement different policies?

_Originally posted by @JenniferHahn in https://github.com/LucasAlegre/sumo-rl/issues/46#issuecomment-1669800758_

**Comment by LucasAlegre** [comment]
Hi @JenniferHahn ,

Thanks for bringing this up! I am not very familiar with the RLLib API, but as far as I remember it is possible to make separate policies for each agent too. Have you checked the examples on their side (e.g., https://github.com/ray-project/ray/blob/master/rllib/examples/multi_agent_independent_learning.py)? I think they address that by changing the policy_mapping_fn.

**Comment by JenniferHahn** [comment]
Thanks a lot for the quick response!! I checked the example and found some other issues when calling the agent_ids with the update policy_mapping_fn but I will figure it out :) 

Quick other question related to this - when I reproduced your script for pop_4x4 with my own network, I always get **0** **number of steps trained,** no matter how much steps sampled. The receive the same output no matter if I use tune or train - I was just wondering if you've ever faced something similar of if you have any idea what might cause this issue?

'counters': {'num_env_steps_sampled': 3584, 'num_env_steps_trained': 0, 'num_agent_steps_sampled': 10752, 'num_agent_steps_trained': 0}

Thanks so much in advance!
